### PR TITLE
Adding multiple processing in source-similarity-matrix.py

### PR DESCRIPTION
**Pull request by aurangzaib048**

### CODE REVIEW

File: source-similarity-matrix.py [hunk 1]
```diff
@@ -9,49 +9,62 @@
...
+                    config.PUB_S3_BUCKET,
+                    f"source-suggestions/{config.SOURCE_EMBEDDINGS.format(LANG_REGION=lang_region)}.csv")
 
 
-def compute_source_similarity(source1, source2, t='dot'):
-    cosine_similarities = np.dot(source1, np.transpose(source2))
-    clip_cosine_similarity = tf.clip_by_value(cosine_similarities, -1.0, 1.0)
-    score = 1.0 - tf.acos(clip_cosine_similarity) / math.pi
-    return score
+def compute_sources_representations(model_lang):
```
Comment (LorenzoMinto) [inline source-similarity-matrix.py:37]: would call this function something like `update_source_sim_files`

File: source-similarity-matrix.py [hunk 2]
```diff
@@ -130,30 +143,31 @@ def get_source_id_for_title(title, sources_df):
...
+    with open(f'output/{config.SOURCE_SIMILARITY_T10.format(LANG_REGION=lang_region)}.json', 'w',
+              encoding='utf-8') as f:
         json.dump(top10_dictionary, f, ensure_ascii=True)
-    with open(f'output/{config.SOURCE_SIMILARITY_T10_HR.format(LANG_REGION=lang_region)}.json', 'w', encoding='utf-8') as f:
+    with open(f'output/{config.SOURCE_SIMILARITY_T10_HR.format(LANG_REGION=lang_region)}.json', 'w',
+              encoding='utf-8') as f:
         json.dump(top10_dictionary_human_readable,
                   f, ensure_ascii=True)
     logger.info("Script has finished running.")
+    upload_source_sim_files(lang_region)
```
Comment (LorenzoMinto) [inline source-similarity-matrix.py:163]: above this line, we could change the log to be a little more informative, i.e. _Finished updating similarity files_.

File: source-similarity-matrix.py [hunk 3]
```diff
@@ -9,49 +9,62 @@
...
+                    f"source-suggestions/{config.SOURCE_EMBEDDINGS.format(LANG_REGION=lang_region)}.csv")
 
 
-def compute_source_similarity(source1, source2, t='dot'):
-    cosine_similarities = np.dot(source1, np.transpose(source2))
-    clip_cosine_similarity = tf.clip_by_value(cosine_similarities, -1.0, 1.0)
-    score = 1.0 - tf.acos(clip_cosine_similarity) / math.pi
-    return score
+def compute_sources_representations(model_lang):
+    def embed(input):
```
Comment (LorenzoMinto) [inline source-similarity-matrix.py:38]: function defs from 38 to 64 could be defined either outside or inside the main function, but I would be consistent with `upload_source_sim_files` and define them all outside (or move that inside as well, but I think outside is more tidy)

Comment (aurangzaib048) [reply source-similarity-matrix.py:38]: The main reason I made them the inner functions is that in multiprocessing every process will have its own memory, and `embed` function is not working outside scope. It was giving me error on line 39.

Comment (LorenzoMinto) [reply source-similarity-matrix.py:38]: I see 👍

File: source-similarity-matrix.py (hunk 2 above)
Comment (aurangzaib048) [reply source-similarity-matrix.py:163]: Done

File: source-similarity-matrix.py (hunk 1 above)
Comment (aurangzaib048) [reply source-similarity-matrix.py]: Done

File: source-similarity-matrix.py [hunk 4]
```diff
@@ -159,15 +159,16 @@ def get_source_id_for_title(title, sources_df):
...
 
 if __name__ == '__main__':
     # Compute similarity matrix for all existing LANG_REGION pairs
     logger.info(f"Number of cpu : {config.CONCURRENCY}")
     with multiprocessing.Pool(config.CONCURRENCY) as pool:
-        tqdm(pool.map(compute_sources_representations, config.LANG_REGION_MODEL_MAP),
+        tqdm(pool.map(update_source_sim_files, config.LANG_REGION_MODEL_MAP),
              total=len(config.LANG_REGION_MODEL_MAP))
 
     logger.info("Completed Sources Similarity")
```
Comment (LorenzoMinto) [inline source-similarity-matrix.py:174]: This is the end of the script, I would say here "script has finished running"

File: source-similarity-matrix.py (hunk 2 above)
Comment (LorenzoMinto) [reply source-similarity-matrix.py:163]: I would change the log to "Updated similarity file for {lang_region}"

**Pull request closed without merging**